src/twister2/specification_processor.py

```diff
@@ -136,6 +144,8 @@ def extract_tests(raw_spec: dict) -> dict:
             if key in test_spec_dict:
                 if key == 'filter':
                     test_spec_dict[key] = _join_filters([test_spec_dict[key], common_value])
+                elif key == 'source_dir':
```

Comment:
Do we need this? (It is handled by `else` statement)

Reply:
Yes, we do.
It would be handled by next elif:
```
  elif isinstance(common_value, str):
      test_spec_dict[key] = _join_strings([test_spec_dict[key], common_value])
```